modules/fitsfilereader/src/renderabletimevaryingfitssphere.cpp

```diff
@@ -0,0 +1,662 @@
+        std::ostringstream oss;
+        oss << digits.substr(0, 4) << "-" // Year
+            << digits.substr(4, 2) << "-" // Month
+            << digits.substr(6, 2) << "T" // Day
+            << digits.substr(8, 2) << ":" // Hour
+            << digits.substr(10, 2) << ":" // Minute
+            << "00"
+            << digits.substr(12, 2) << "." // Second
+            << "000";
+
+        return openspace::Time::convertTime(oss.str());
+    }
+
+    constexpr openspace::properties::Property::PropertyInfo TextureSourceInfo = {
+        "TextureSource",
+        "Texture Source",
+        "This value specifies a directory from where files are loaded from "
+        "disk and are used as a texture that is applied to this sphere.",
+        openspace::properties::Property::Visibility::User
+    };
+
+    constexpr openspace::properties::Property::PropertyInfo FitsLayerInfo = {
+        "FitsLayer",
+        "Texture Layer",
+        "This value specifies which index in the fits file to use as texture.",
+        openspace::properties::Property::Visibility::User
+    };
+
+    constexpr openspace::properties::Property::PropertyInfo FitsLayerNameInfo = {
+        "LayerNames",
+        "Texture Layer Options",
+        "This value specifies which name of the fits layer to use as texture.",
+        openspace::properties::Property::Visibility::User
+    };
+
+    constexpr openspace::properties::Property::PropertyInfo TextureFilterInfo = {
+        "TextureFilter",
+        "Texture Filter",
+        "Option to choose nearest neighbor or linear filtering for the texture.",
+        openspace::properties::Property::Visibility::User
+    };
+
+    constexpr openspace::properties::Property::PropertyInfo SaveDownloadsOnShutdown = {
+        "SaveDownloadsOnShutdown",
+        "Save Downloads On Shutdown",
+        "This is an option for if dynamically downloaded should be saved between runs "
+        "or not.",
+        openspace::properties::Property::Visibility::User
+    };
+
+    struct [[codegen::Dictionary(RenderableTimeVaryingFitsSphere)]] Parameters {
+        // [[codegen::verbatim(TextureSourceInfo.description)]]
+        std::optional<std::filesystem::path> textureSource;
+
+        enum class [[codegen::map(
+            openspace::RenderableTimeVaryingFitsSphere::LoadingType)]] LoadingType
+        {
+            StaticLoading,
+            DynamicDownloading
+        };
+
+        // Choose type of loading:
+        // StaticLoading: Download and load files on startup.
+        // DynamicDownloading: Download and load files during run time.
+        std::optional<LoadingType> loadingType;
+
+        // A data ID that corresponds to what dataset to use if using dynamicWebContent.
+        std::optional<int> dataID;
+
+        // This is a max value of the amount of files to queue up
+        // so that not to big of a data set is downloaded.
+        std::optional<int> numberOfFilesToQueue;
```

Comment:
See previous suggestion for rewrite in RenderableFieldLinesSequence

Almost looks like it could have been useful to make a component for the dynamic downloading, that collects all these settings so that they do not have to be duplicated for the two renderables that use them. 

See for example `ColorMappingComponent`. But I understand if this is too much of a change for this PR